gin/config.py

```diff
@@ -248,12 +248,17 @@ def _decorate_fn_or_cls(decorator, fn_or_cls, subclass=False):
   construction_fn = _find_class_construction_fn(fn_or_cls)
 
   if subclass:
-
+    klass = fn_or_cls
+    while hasattr(klass, '_gin_decorated') and klass._gin_decorated:
+      klass = klass.__bases__[0]
     class DecoratedClass(fn_or_cls):
       __doc__ = fn_or_cls.__doc__
       __module__ = fn_or_cls.__module__
-
+      __class__ = klass
```

Comment:
I wonder if we could remove the "_gin_decorated" attribute and associated base class traversal above if we use `__class__ = fn_or_cls.__class__` here? I think that would propagate the right `__class__` down even when decorated multiple times (although haven't tested).

Reply:
I've tested that (or at least what I think you think of) and it didn't work so smoothly for external configurables 🙂